test-ciphers.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+console_url=$(oc get console.config.openshift.io cluster --template '{{.status.consoleURL}}')
+if [ -z "${console_url}" ]
+then
+  echo "no console_url, are you connected to a cluster?"
+  exit 1
+fi
+
+# need to format the console_url for s_client
+console_url_without_http=${console_url#"https://"}
+console_url_without_http_with_port="${console_url_without_http}:443"
+SERVER="${console_url_without_http_with_port}"
+
+
+# ensure what we expect works
+# we approve roughly 16 cipher suites
+# but
+valid_cipher_sample=(
+  ECDHE-ECDSA-CHACHA20-POLY1305
+  ECDHE-RSA-AES256-GCM-SHA384
+)
+
+echo "testing valid ciphers...."
+# we want these to successfully connect
+for CIPHER in "${valid_cipher_sample[@]}"
+do
+  # test one, silencing output
+  result=$(echo -n | openssl s_client -connect $SERVER -cipher "${CIPHER}" 2>&1)
+  if [[ "$result" =~ "no cipher match" ]] ; then
+    echo "valid cipher suite was denied (${CIPHER})"
+    exit 1
+  fi
+done
+
+# ensure we ignore weak ciphers
+# CBC (cipher block chaining) are no longer reliable and should not be used
+# CBC ciphers use an IV (initialization vector) and a chaining mechanism.
+# The chaining mechanism means that a single bit error in a ciphertext block
+# will invalidate all previous blocks.  The chaining was good in that it hides
+# plaintext patterns, but is inferior to other cipher modes.
+invalid_cipher_sample=(
+  RSA-AES-128-CBC-SHA256
+  ECDHE-RSA-3DES-EDE-CBC-SHA    # disabled to mitigate SWEET32 attack
+  RSA-3DES-EDE-CBC-SHA          # disabled to mitigate SWEET32 attack
+)
+
+echo "testing invalid ciphers...."
+# we want these to fail
+for CIPHER in "${invalid_cipher_sample[@]}"
+do
+  # test one, silencing output
+  result=$(echo -n | openssl s_client -connect $SERVER -cipher "${CIPHER}" 2>&1)
```

Comment:
This should be fine, but it's a good practice to use quotes here anyway

```suggestion
  result=$(echo -n | openssl s_client -connect "$SERVER" -cipher "${CIPHER}" 2>&1)
```